jupyterhub_singleuser_profiles/service.py

```diff
@@ -37,91 +39,91 @@ def __init__(self, server_url, token, namespace=None, verify_ssl=True):
-    except Exception:
-        _LOGGER.error("Failed to process template: %s", response.text)
-        raise
+    tmp = jinja2.Template(json.dumps(template))
 
-    processed = response.json()
-    if len(processed["objects"]) != 1:
-      raise Exception("Template must contain a single ConfigMap or CustomResource")
+    result = tmp.render(configuration)
 
-    configMap = processed["objects"][0]
-    if not configMap["metadata"].get("labels"):
-      configMap["metadata"]["labels"] = {}
+    result = json.loads(result)
+    if not result.get('metadata'):
+      result['metadata'] = {}
+    if labels:
+      result['metadata']['labels'] = labels
+    
+    result['metadata']['name'] = service_name + '-' + user
 
-    configMap["metadata"]["labels"][_SERVICE_LABEL] = escape(user)
+    return result
 
-    return configMap
 
-  def submit_resource(self, resource, return_paths):
-    if resource["kind"] == "ConfigMap":
-      name = resource["metadata"]["name"]
-      try: 
-        api_response = self.k8s_api_instance.replace_namespaced_config_map(name, self.namespace, resource, pretty="true", )
-        result = self._get_data_from_response(api_response, return_paths)
-        return result
-      except ApiException as e:
-        try:
-          if e.status == 404:
-            api_response = self.k8s_api_instance.create_namespaced_config_map(self.namespace, resource, pretty="true")
-            result = self._get_data_from_response(api_response, return_paths)
-            return result
-          else:
-            _LOGGER.error("Exception when calling CoreV1Api->replace_namespaced_config_map: %s\n" % e)
-            raise
-        except ApiException as e2:
-          _LOGGER.error("Exception when calling CoreV1Api->create_namespaced_config_map: %s\n" % e2)
+  def submit_resource(self, processed_template, return_paths):
+    client_wrapper = self.os_client.resources.get(api_version=processed_template['apiVersion'], kind=processed_template['kind'])
+    try:
+      response = client_wrapper.create(body=processed_template, namespace=self.namespace)
+      result = self._get_data_from_response(response, return_paths)
+      return result
+    except ApiException as e:
+      try:
+        if e.status == 409:
+          # Added because of https://github.com/kubernetes/kubernetes/issues/70674#issuecomment-438569688
+          original_resource = client_wrapper.get(name=processed_template['metadata']['name'], namespace=self.namespace)
+          original_dict = original_resource.to_dict()
+          processed_template['metadata']['resourceVersion'] = original_dict['metadata']['resourceVersion']
+          response = client_wrapper.replace(body=processed_template, namespace=self.namespace)
+          result = self._get_data_from_response(response, return_paths)
+
+          # Necessary for deletion purposes
+          support_confm = self.get_support_config_map()
+          support_confm['data'][processed_template['metadata']['name']] = {
```

Comment:
I'd suggest the structure for CM to be

```
data:
  resources: |
    - apiVersion: ...
      kind: ...
      name:  ...
```

from the implementation perspective `resources` is a key in `data` and the content of `resources` is `string`, so you need to handle yaml (de)serialization